gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -106,14 +131,47 @@ public DagTask next() {
     return null;
```

Comment:
probably best to put the `try / catch` within the `while` loop.  that way no occasionally returning `null`.

also, should we really catch `Throwable`?  seems extreme.  maybe just `Exception`?